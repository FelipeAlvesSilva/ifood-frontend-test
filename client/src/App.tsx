import React from 'react';

function App() {
  return (
   <div>
     <h2>Spotifood</h2>
   </div>
  );
}

export default App;
