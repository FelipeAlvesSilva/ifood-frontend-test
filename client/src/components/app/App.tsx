import React from 'react';
import { withTheme } from 'styled-components';
import { TiWeatherPartlySunny as Sun } from 'react-icons/ti';
import { GiMoon as Moon } from 'react-icons/gi';
import { Switch } from 'antd';
import { appTheme } from '../theme/ThemeContext';

function App() {
  const themeToggle = appTheme();

  return (
    <div>
      <Switch
        onClick={() => themeToggle.toggle()}
        checkedChildren={<Moon />}
        unCheckedChildren={<Sun />}
        defaultChecked
      />
    </div>
  );
}

// HOC to obtain theme prop
export default withTheme(App);
