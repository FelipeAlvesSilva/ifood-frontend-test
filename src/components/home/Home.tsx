import React, { Fragment } from 'react';
import { withTheme } from 'styled-components';
import { FaSearch } from 'react-icons/fa';
import { AppTheme } from '../theme/ThemeContext';
import { CInput as Input } from '../styled/Input';
import HeaderNav from '../header/HeaderNav';
import Filters from '../filters/Filters';

const Home = (props) => {
  // const [searchValue, setSearchValue] = useState('');
  const themeToggle = AppTheme();

  return (
    <Fragment>
      <HeaderNav toggle={themeToggle.toggle} />
      <Input
        className="m-3"
        // onChange={(e) => setSearchValue(e.target.value)}
        placeholder="Search..."
        prefix={<FaSearch className="mb-2 mt-2 mr-3 ml-2" />}
      />
      <Filters />
    </Fragment>
  );
}

// HOC to obtain theme prop
export default withTheme(Home);
