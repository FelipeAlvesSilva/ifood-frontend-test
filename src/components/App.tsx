import React, { Fragment } from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import Home from './home/Home';
import Logon from './logon/Logon';
import Fail from './fail/Fail';

const App = () => {
  return (
    <Fragment>
      <Router>
        <Switch>
          <Route path="/" exact component={Logon} />
          <Route path="/home" exact component={Home} />
          <Route path="/fail" exact component={Fail} />
          <Redirect from="*" to="/fail" />
        </Switch>
      </Router>
    </Fragment>
  )
};

export default App;