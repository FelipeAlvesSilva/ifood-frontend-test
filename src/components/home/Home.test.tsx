import React from 'react';
import { render } from '@testing-library/react';
import Home from './Home';

// test('renders learn react link', () => {
//   const { getByText } = render(<Home />);
//   const linkElement = getByText(/spotifood/i);
//   expect(linkElement).toBeInTheDocument();
// });


/**
 * shallow option
 * 
 *  import React from 'react';
    import { shallow } from 'enzyme';
    import App from './App';
    it('renders welcome message', () => {
      const wrapper = shallow(<App />);
      const welcome = <h2>Welcome to React</h2>;
      expect(wrapper.contains(welcome)).toEqual(true);
    });
 */