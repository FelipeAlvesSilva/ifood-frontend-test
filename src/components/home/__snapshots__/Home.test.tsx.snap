// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Home Component Home Component should not regress 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container my-4"
      >
        <h1
          class="sc-AxirZ ixxgie m-5"
        >
          <img
            alt="Spotify brand icon"
            class="mr-3"
            src="Spotify_Icon_CMYK_Green.png"
            style="height: 60px; width: 60px;"
          />
          <a
            class="custom-link"
            href="/home"
          >
            Spotifood
          </a>
          <button
            aria-checked="true"
            class="ant-switch ant-switch-checked"
            role="switch"
            style="float: right;"
            type="button"
          >
            <div
              class="ant-switch-handle"
            />
            <span
              class="ant-switch-inner"
            >
              <svg
                class="switcher-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M195 125c0-26.3 5.3-51.3 14.9-74.1C118.7 73 51 155.1 51 253c0 114.8 93.2 208 208 208 97.9 0 180-67.7 202.1-158.9-22.8 9.6-47.9 14.9-74.1 14.9-106 0-192-86-192-192z"
                />
              </svg>
            </span>
          </button>
        </h1>
      </div>
      <span
        class="sc-AxjAm iJLKWD m-2 ant-input-affix-wrapper"
      >
        <span
          class="ant-input-prefix"
        >
          <svg
            class="mb-2 mt-2 mr-3 ml-2"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
        </span>
        <input
          class="ant-input"
          placeholder="Search..."
          type="text"
          value=""
        />
      </span>
      <div
        class="ant-collapse ant-collapse-borderless ant-collapse-icon-position-right m-3"
      >
        <div
          class="ant-collapse-item"
        >
          <div
            aria-expanded="false"
            class="ant-collapse-header"
            role="button"
            tabindex="0"
          >
            <span
              aria-label="right"
              class="anticon anticon-right ant-collapse-arrow"
              role="img"
            >
              <svg
                aria-hidden="true"
                class=""
                data-icon="right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
                />
              </svg>
            </span>
            Filters
            <div
              class="ant-collapse-extra"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 16 16"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M6 10.5a.5.5 0 01.5-.5h3a.5.5 0 010 1h-3a.5.5 0 01-.5-.5zm-2-3a.5.5 0 01.5-.5h7a.5.5 0 010 1h-7a.5.5 0 01-.5-.5zm-2-3a.5.5 0 01.5-.5h11a.5.5 0 010 1h-11a.5.5 0 01-.5-.5z"
                  fill-rule="evenodd"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container my-4"
    >
      <h1
        class="sc-AxirZ ixxgie m-5"
      >
        <img
          alt="Spotify brand icon"
          class="mr-3"
          src="Spotify_Icon_CMYK_Green.png"
          style="height: 60px; width: 60px;"
        />
        <a
          class="custom-link"
          href="/home"
        >
          Spotifood
        </a>
        <button
          aria-checked="true"
          class="ant-switch ant-switch-checked"
          role="switch"
          style="float: right;"
          type="button"
        >
          <div
            class="ant-switch-handle"
          />
          <span
            class="ant-switch-inner"
          >
            <svg
              class="switcher-icon"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M195 125c0-26.3 5.3-51.3 14.9-74.1C118.7 73 51 155.1 51 253c0 114.8 93.2 208 208 208 97.9 0 180-67.7 202.1-158.9-22.8 9.6-47.9 14.9-74.1 14.9-106 0-192-86-192-192z"
              />
            </svg>
          </span>
        </button>
      </h1>
    </div>
    <span
      class="sc-AxjAm iJLKWD m-2 ant-input-affix-wrapper"
    >
      <span
        class="ant-input-prefix"
      >
        <svg
          class="mb-2 mt-2 mr-3 ml-2"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
      </span>
      <input
        class="ant-input"
        placeholder="Search..."
        type="text"
        value=""
      />
    </span>
    <div
      class="ant-collapse ant-collapse-borderless ant-collapse-icon-position-right m-3"
    >
      <div
        class="ant-collapse-item"
      >
        <div
          aria-expanded="false"
          class="ant-collapse-header"
          role="button"
          tabindex="0"
        >
          <span
            aria-label="right"
            class="anticon anticon-right ant-collapse-arrow"
            role="img"
          >
            <svg
              aria-hidden="true"
              class=""
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
          Filters
          <div
            class="ant-collapse-extra"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                clip-rule="evenodd"
                d="M6 10.5a.5.5 0 01.5-.5h3a.5.5 0 010 1h-3a.5.5 0 01-.5-.5zm-2-3a.5.5 0 01.5-.5h7a.5.5 0 010 1h-7a.5.5 0 01-.5-.5zm-2-3a.5.5 0 01.5-.5h11a.5.5 0 010 1h-11a.5.5 0 01-.5-.5z"
                fill-rule="evenodd"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
